admin_solutions/authentication.adoc

```diff
@@ -0,0 +1,166 @@
+For now, this topic focuses on GitHub as a source of authentication, but more
+providers will be added in a future release.
+
+[[github-auth]]
+== GitHub Authentication
+
+GitHub uses OAuth, and you can integrate your OpenShift cluster to use that
+OAuth authentication. OAuth basically facilitates a token exchange flow.
+
+Configuring GitHub authentication allows users to log in to OpenShift with their
+GitHub credentials. To prevent anyone with a GitHub user ID to log in to your
+OpenShift cluster, you can restrict users to require their membership in GitHub
+organizations.
+
+[[register-app-on-github]]
+=== Registering the Application on GitHub
+
+Before you do an integration like this, you must be a registered client on the
+remote service:
+
+. On GitHub, click https://github.com/settings/profile[Settings] ->
+https://github.com/settings/applications[OAuth applications] ->
+https://github.com/settings/developers[Developer applications] ->
+https://github.com/settings/applications/new[Register an application]
+to navigate to the page for a
+https://github.com/settings/applications/new[new OAuth application].
+. Type an application name. For example: `My OpenShift Install`
+. Type a homepage URL. For example: `https://myapiserver.com:8443`
+. Optionally, type an application description.
+. Type the authorization callback URL, where the end of the URL contains the
+identity provider *name* (defined in the `*identityProviders*` stanza of the link:../install_config/master_node_configuration.html[*_master configuration file_*], which you configure in the next section of this topic):
++
+<apiserver>/oauth2callback/<identityProviderName>
++
+For example:
++
+https://myapiserver.com:8443/oauth2callback/github/
+. Click *Register application*. GitHub provides a Client ID and a Client Secret.
+Keep this window open so you can copy these value and paste them into the
+master configuration file.
+
+[[configuring-auth-on-master]]
+=== Configuring Authentication on the Master
+
+. If you have:
+.. Already completed installation of Openshift, then copy the
+*_master-config.yaml_* file into a new directory; for example:
++
+----
+$ mkdir githubconfig; cp master-config.yaml githubconfig
+----
+.. Not yet installed OpenShift, then start the OpenShift API server, specifying
+the hostname of the (future) OpenShift master and a directory to store the
+configuration file created by the start command:
++
+----
+$ openshift start master --public-master=<apiserver> --write-config=<directory>
+----
++
+For example:
++
+----
+$ openshift start master --public-master=https://myapiserver.com:8443 --write-config=githubconfig
+----
++
+[NOTE]
+====
+If you are installing by Ansible, then you must provide Ansible with the
+identityProvider configuration. When installing by Ansible, you probably do not
+want to be manually modifying your configuration, because you would lose the
+modifications when you re-ran your install, or upgraded. This topic shows you
```

Comment:
I think the above sentence might be a bit unnecessarily wordy... Maybe:

"Modifying your configuration manually after installing with Ansible results in losing any modifications when re-running the install tool or upgrading."